### PR TITLE
fix a flake in the sync manager edge case test

### PR DESCRIPTION
**Pull request by Stebalien**
Depending on scheduling, we may see c2 before we see c1.

### CODE REVIEW

**Review by schomatis — COMMENTED** [review]
LGTM I can't reproduce locally (after 1000's of runs) but this seems like it should fix the issue.

I'm deferring to @vyzo for the actual approval; accepting either `c2` or `c1` seems to defeat the purpose of the test altogether. I'm not sure how the `done()` API works but, as `Sync()` can in theory be run in parallel, to process tipsets in the proper order I suppose that this test is working under the assumption that `syncManager` is going to throttle them accordingly (coming from the same parent). If this is not the case then I have no idea what this test is doing and my green tick shouldn't count anyways.

**Comment by Stebalien** [comment]
I _believe_ the goal is that c1 should be the final decision, even if c1 comes first. This still tests that (ish).

**Review by schomatis — APPROVED** [review]
Looking at this again, `c1` and `c2` do _not_ come from the same parent as I originally thought. `Sync()` calls will be issued in parallel for both, so this is in fact racy (not sure if by design). The fix makes sense as a temporary solution.

**Review by vyzo — APPROVED** [review]
yes, that's the intention! thanks for fixing this.

**Review by magik6k — APPROVED** [review]
:heart:

**Pull request merged**
